src/shared/copy.c

```diff
@@ -489,8 +489,12 @@ static int fd_copy_symlink(
         r = symlinkat(target, dt, to);
         if (copy_flags & COPY_MAC_CREATE)
                 mac_selinux_create_file_clear();
-        if (r < 0)
+        if (r < 0) {
+                if (FLAGS_SET(copy_flags, COPY_SYMLINKS_GRACEFUL) && (ERRNO_IS_PRIVILEGE(r) || ERRNO_IS_NOT_SUPPORTED(r)))
```

Comment:
please add a debug message at least

Reply:
you are checking `r` here, even though the error code is actually in errno here.

maybe use RET_NERRNO() around symlinkat() to avoid this confusion for good.

Reply:
actually, thinking about this maybe rename the flag COPY_SYMLINKS_GRACEFUL_WARN and then log about this thing via log_notice(). This actually appears to be something people should know about when it happens.